### PR TITLE
Replace uses of Poperty<File> with FileProperty in plugins

### PR DESCRIPTION
**Pull request by rlazo**
This is the recommended approach by Gradle

https://docs.gradle.org/current/userguide/properties_providers.html#managed_properties

### CODE REVIEW

File: buildSrc/src/main/java/com/google/firebase/gradle/plugins/DackkaPlugin.kt [hunk 1]
```diff
@@ -204,26 +204,26 @@ abstract class DackkaPlugin : Plugin<Project> {
...
   private fun registerFiresiteTransformTask(
     project: Project,
-    dackkaOutputDirectory: Provider<File>,
+    dackkaOutputDirectory: Provider<RegularFile>,
   ) =
     project.tasks.register<FiresiteTransformTask>("firesiteTransform") {
       val outputDir by tempFile("dackkaTransformedFiles")
 
-      dackkaFiles.set(dackkaOutputDirectory.childFile("docs/reference"))
+      dackkaFiles.set(dackkaOutputDirectory.get().asFile.childFile("docs/reference"))
```
Comment (daymxn) [inline buildSrc/src/main/java/com/google/firebase/gradle/plugins/DackkaPlugin.kt:220]: `.get()` eagerly resolves the provider. You'll wanna `map` it instead.

Comment (rlazo) [reply buildSrc/src/main/java/com/google/firebase/gradle/plugins/DackkaPlugin.kt]: done

**Review by daymxn — CHANGES_REQUESTED** [review]
Something is likely wrong with the api generation, see the test failures for context.

**Comment by rlazo** [comment]
> Something is likely wrong with the api generation, see the test failures for context.

Addressed

**Review by daymxn — APPROVED** [review]

**Pull request merged**